src/rest.cpp

```diff
@@ -950,6 +998,8 @@ static const struct {
       {"/rest/mempool/contents", rest_mempool_contents},
       {"/rest/headers/", rest_headers},
       {"/rest/getutxos", rest_getutxos},
+      {"/rest/deploymentinfo/", rest_deploymentinfo},
+      {"/rest/deploymentinfo", rest_deploymentinfo},
```

Comment:
Why this api supports both `/rest/deploymentinfo` and `/rest/deploymentinfo/` while others not. Is this a good idea to add the both to other apis?

Reply:
This API supports both because `getdeploymentinfo` can be used by passing or not passing a block hash. So, you can use: `/rest/deploymentinfo/BLOCKHASH.json` or `/rest/deploymentinfo.json` if you want to get the information from the current chain tip.

For other endpoints, passing a BLOCKHASH or a TXID is a must which is not this case.

Reply:
+0 on this. I don't think it would be a bad practice, in a user perspective I'd prefer:
`/rest/deploymentinfo.json` to get all and `/rest/deploymentinfo/<BLOCKHASH>.json` to get a specific one. I don't feel myself (as a final user) confortable with `/rest/deploymentinfo/.json`. 

I agree 100000% with you the format parameter should be a query one, and if you're working on this, maybe we can keep it this way and change it after your PR?

Reply:
If no one else thinks this is an issue I'm okay to proceed with it, but having something like `GET /rest/deploymentinfoa.json` resolve to a blockhash of `a` does not sit well with me. Either way, it doesn't impact the upgrade path with query params I have in mind so from that end I'm not too bothered.